plutus-core/plutus-core/src/PlutusCore/Flat.hs

```diff
@@ -363,3 +364,15 @@ deriving newtype instance Flat (Binder TyName)
 -- That would be more compact, but we don't need it at the moment.
 deriving newtype instance Flat (Binder NamedDeBruijn)
 deriving newtype instance Flat (Binder NamedTyDeBruijn)
+
+-- this instance is very similar to the Flat DeBruijn instance.
+instance Flat FakeNamedDeBruijn where
+    size = error "we only use fakenameddebruijn for decoding optimization"
+    encode = error "we only use fakenameddebruijn for decoding optimization"
+    decode =  toFake <$> decode -- via debruijn
+
+-- this instance is very similar to the Flat (Binder DeBruijn) instance.
+instance Flat (Binder FakeNamedDeBruijn) where
+    size = error "we only use fakenameddebruijn for decoding optimization"
+    encode = error "we only use fakenameddebruijn for decoding optimization"
+    decode = coerce . toFake . coerce <$> decode @(Binder DeBruijn) -- via binder debruijn
```

Comment:
@michaelpj Do you think it is worth to build a specialized decoder just for ScriptForExecution (via parameterized `decodeterm` and some boilerplate `Flat.Get` functions) instead of relying on these partial Flat.Class instances? 

Or perhaps are you ok with these non-roundtripping flat instances as long as they are not used?

Reply:
Um. I guess I would probably fill in the other methods too? We *can* write a roundtripping instance *provided* the invariants of `FakeNameDeBruijn` are upheld (i.e. the name is only ever `fakeName`).